crypto/mem_sec.c

```diff
@@ -78,9 +73,11 @@ int CRYPTO_secure_malloc_init(size_t size, int minsize)
 void CRYPTO_secure_malloc_done()
 {
 #ifdef IMPLEMENTED
-    sh_done();
-    secure_mem_initialized = 0;
-    CRYPTO_THREAD_lock_free(sec_malloc_lock);
+    if (secure_mem_used == 0) {
```

Comment:
I was trying to avoid additional API changes, but void -> int is probably harmless, in this case.